danielMunteanu/src/main/java/danielCollectionTask/AbstractOrder.java

```diff
@@ -0,0 +1,67 @@
+package danielCollectionTask;
+
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.UUID;
+
+public abstract class AbstractOrder implements Order {
+    final String id;
+    final LocalDateTime date;
+    private List<Item> items;
+
+    public AbstractOrder(List<Item> items) {
+        this.id = id();
+        this.date = LocalDateTime.now();
+        this.items = items;
+    }
+
+    private static String id() {
+        return String.valueOf(UUID.randomUUID());
+    }
+
+    public String getId() {
+        return id;
+    }
+
+    public LocalDateTime getDate() {
+        return date;
+    }
+
+    public List<Item> getItems() {
+        return items;
+    }
+
+    public void setItems(List<Item> items) {
```

Comment:
Should we have such method as time as we already are providing the items list by calling the Constructor?

Reply:
yes, otherwise the program will not work

Reply:
are you sure it will not work?

Reply:
I'm using it to create user orders 
![image](https://user-images.githubusercontent.com/62387822/78762813-0d608380-798d-11ea-9a43-e921a434b790.png)
![image](https://user-images.githubusercontent.com/62387822/78762837-15202800-798d-11ea-85fd-0eaed8fdaaa1.png)

Reply:
There is a problem, you are setting the list with the help of constructor, after that you set it one more time with the help of setItems() method, it's not a good approach. Use only one way to do it.

Reply:
got you,setItems removed